dev-util/colm/colm-0.13.0.5.recipe

```diff
@@ -0,0 +1,79 @@
+SUMMARY="COmputer Language Manipulation"
+DESCRIPTION="Colm is a programming language designed for the analysis and \
+transformation of computer languages. Colm is influenced primarily by TXL. It \
+is in the family of program transformation languages."
+HOMEPAGE="https://www.colm.net/open-source/colm/"
+COPYRIGHT="2006-2016 Adrian Thurston"
+LICENSE="MIT"
+REVISION="1"
+SOURCE_URI="https://www.colm.net/files/colm/colm-$portVersion.tar.gz"
+CHECKSUM_SHA256="33e624677176958eaad76ebe6c391a68a0b4728fec8cc039efa1316f525f408c"
+
+ARCHITECTURES="?x86_gcc2 ?x86 x86_64"
+SECONDARY_ARCHITECTURES="?x86"
+
+if [ "$targetArchitecture" != x86_gcc2 ]; then
+	commandSuffix=$secondaryArchSuffix
+	commandBinDir=$binDir
+else
+	commandSuffix=
+	commandBinDir=$prefix/bin
+fi
```

Comment:
What do you think about this variation? This one will disable the suffix whenever it couldn't be built on the main architecture.

```sh
commandSuffix=$secondaryArchSuffix
commandBinDir=$binDir
if [ -z "${ARCHITECTURES##*!$targetArchitecture*}" ]; then
	commandSuffix=
	commandBinDir=$prefix/bin
fi

Reply:
Hmm, it's not very eye-friendly.
What I recommend is the more traditional style:
```
commandBinDir=$binDir
commandSuffix=$secondaryArchSuffix
if [ "$targetArchitecture" = x86_gcc2 ]
then
	commandBinDir=$prefix/bin
	commandSuffix=
fi
```

Reply:
Given that probably only `x86_gcc2` would need this, I think it's fine to only use this for said architecture.

BTW, should we add this to HaikuPorter itself? Since we don't want those suffixes for binaries when a port won't build for the main architecture anyway.

Reply:
1) +1, because x86_gcc2 is, indeed, the only arch for which we use `commandBinDir`.
(We would have been able to use it for old apps which only build gcc2, if we had x86_gcc2 as second arch for x86_64 primary arch, but we don't build images with that combination.)

2) I'd say it is better to leave that outside of HaikuPorter. This gives more flexibility, in particular if we want to allow users of x86_gcc2 hybrid to have the `/system/bin/foobar` from the primary arch package and the `/system/bin/x86/foobar` from the secondary arch. We generally choose to drop the latter when the former works fine, but we could have situations where we're not sure the runtime built with gcc2 works perfectly.

Reply:
In fact there are special cases already. For example:
 * ffmpeg builds with both archs but we ship the binaries from the x86 version, because they work better.
 * For gcc (and most tools in the build toolchain), we do want to ship both versions.

If there was an automated solution to this, we would not have introduced an explicit $secondaryArchSuffix at all (at least for binaries).